src/components/Footer/Footer.js

```diff
@@ -53,23 +76,99 @@ const Footer = ({
     }
     return '#C4C4C4';
   };
+
+  const username = item?.anon_user_info_emoji_name
+    ? `Anonymous ${item?.anon_user_info_emoji_name}`
+    : item?.actor?.data?.username;
+
+  const onPressDM = async () => {
+    try {
+      setLoading({...loading, loadingDm: true});
+      if (!item?.anon_user_info_emoji_name) {
+        const channelName = [username, profile?.myProfile?.username].join(',');
+        const selectedUser = {
+          user: {
+            name: channelName,
+            image: item?.actor?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
+          }
+        };
+        const members = [item?.actor?.id, profile.myProfile.user_id];
+        await createSignChat(members, selectedUser);
+      } else {
+        await sendMessageDM(item.id, 'post', 'SIGNED');
+      }
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDm: false});
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `onPressDM` function, there's a console warning when an error occurs. This is fine for development, but in production, you should consider using a logging service or some other form of error reporting. Also, the error handling does not provide any user feedback. Consider providing a user-friendly message if an error occurs.
```javascript
try {
  // existing code...
} catch (e) {
  console.warn(e);
  // consider adding user feedback here, e.g.:
  // showNotification('An error occurred while sending the message.');
} finally {
  // existing code...
}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->